src/python/WMCore/MicroService/MSUnmerged/MSUnmerged.py

```diff
@@ -319,120 +332,92 @@ def cleanRSE(self, rse):
             self.logger.exception(msg)
             raise MSUnmergedPlineExit(msg) from ex
 
-        filesToDeleteCurrRSE = 0
-
         # Start cleaning one directory at a time:
-        for dirLfn, fileLfnGen in rse['files']['toDelete'].items():
-            if dirLfn in rse['dirs']['deletedSuccess']:
-                self.logger.info("RSE: %s, dir: %s already successfully deleted.", rse['name'], dirLfn)
-                continue
+        for idx, dirLfn in enumerate(rse['dirs']['toDelete']):
```

Comment:
As far as I can tell the `toDelete` parts of the `rse['dirs]` object is never cleaned up even though the logic below deletes dirs/files.

Reply:
Instead of loop I suggest to clearly define function(s) to delete an object. The function can take `fobj` (either dir or file) and perform the operation and return the status. Then code can be parallelized (to speed up nested operations) and deletion can be done concurrently.